lua/zk/adapter.lua

```diff
@@ -1,12 +1,55 @@
 -- The interface between our plugin and zk
 
-local adapter = {}
+local M = {}
 
-function adapter.new()
+function M.new(title)
+  local cmd = "zk new"
+
+  if title ~= nil and title ~= "" then
+    cmd = string.format("zk new --title %s", title)
```

Comment:
seems like we don't care about further arguments passed in, just a title is sufficient as an optional arg, yeah?

Reply:
The destination directory and/or `--group` could be useful too, to generate different kind of notes. The group is actually derived from the destination directory. See [this note](https://github.com/mickael-menu/zk/blob/main/docs/config-group.md) and `zk new --help`.

You can also feed some initial content to the note, which will be expanded in the template with `{{content}}`. This needs to be fed through an stdin pipe. Maybe it's overkill for now, but could be nice later to create a new note from the current buffer selection, to split a big note into smaller ones.

Reply:
> The destination directory and/or `--group` could be useful too, to generate different kind of notes. The group is actually derived from the destination directory. See [this note](https://github.com/mickael-menu/zk/blob/main/docs/config-group.md) and `zk new --help`.
> 
> You can also feed some initial content to the note, which will be expanded in the template with `{{content}}`. This needs to be fed through an stdin pipe. Maybe it's overkill for now, but could be nice later to create a new note from the current buffer selection, to split a big note into smaller ones.

I love this idea for the note management aspect. Adding for future enhancement idea.